gnomad/sample_qc/pipeline.py

```diff
@@ -479,11 +483,18 @@ def annotate_sex(
         if variants_only_y_ploidy:
             ploidy_ht = ploidy_ht.drop("chrY_ploidy", "chrY_mean_dp")
 
-    if var_keep_contigs:
+    add_globals = hl.struct()
+    if compute_x_frac_variants_hom_alt or var_keep_contigs:
         logger.info(
-            "Filtering variants for variant only sex chromosome ploidy imputation."
+            "Filtering variants for variant only sex chromosome ploidy imputation and/or computation of the fraction "
+            "of homozygous alternate variants on chromosome X",
         )
-        filtered_mt = hl.filter_intervals(mt, var_keep_locus_intervals)
+        if variants_only_x_ploidy:
+            filtered_mt = hl.filter_intervals(mt, var_keep_locus_intervals)
```

Comment:
functionally speaking is line 493 and 496 doing the same thing?

Reply:
Oh good point, I can just do and it will work for both
```
filtered_mt = hl.filter_intervals(
                mt, var_keep_locus_intervals + x_locus_intervals
            )
```